<?xml version="1.0" encoding="UTF-8"?>
<testng-results retried="10" ignored="0" total="40" passed="25" failed="5" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-08-28T11:50:33 EDT" name="Suite" finished-at="2025-08-28T11:52:58 EDT" duration-ms="145051">
    <groups>
      <group name="datadriven">
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" name="testLogin" class="testCases.TC02_Login"/>
      </group> <!-- datadriven -->
      <group name="regression">
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@22875539]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@e54303]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
        <method signature="TC05_AddToWishList.testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" name="testAddToWishList" class="testCases.TC05_AddToWishList"/>
        <method signature="TC06_AddAffiliate.testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@2dc995f4]" name="testAddAffiliate" class="testCases.TC06_AddAffiliate"/>
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@49c6c24f]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@6fe1b4fb]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
        <method signature="TC05_AddToWishList.testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" name="testAddToWishList" class="testCases.TC05_AddToWishList"/>
        <method signature="TC06_AddAffiliate.testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@1c32386d]" name="testAddAffiliate" class="testCases.TC06_AddAffiliate"/>
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@5d5f10b2]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@74c79fa2]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@3e30646a]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
        <method signature="TC05_AddToWishList.testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" name="testAddToWishList" class="testCases.TC05_AddToWishList"/>
        <method signature="TC06_AddAffiliate.testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@53bd8fca]" name="testAddAffiliate" class="testCases.TC06_AddAffiliate"/>
      </group> <!-- regression -->
      <group name="sanity">
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@22875539]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@e54303]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@49c6c24f]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@6fe1b4fb]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
        <method signature="TC01_LaunchApplication.testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@5d5f10b2]" name="testLaunchApplication" class="testCases.TC01_LaunchApplication"/>
        <method signature="TC02_Login.testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" name="testLogin" class="testCases.TC02_Login"/>
        <method signature="TC03_AddToCart.testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@74c79fa2]" name="testAddToCart" class="testCases.TC03_AddToCart"/>
        <method signature="TC04_CompletePurchase.testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@3e30646a]" name="testCompletePurchase" class="testCases.TC04_CompletePurchase"/>
      </group> <!-- sanity -->
    </groups>
    <test started-at="2025-08-28T11:50:33 EDT" name="edgeTest" finished-at="2025-08-28T11:52:21 EDT" duration-ms="108556">
      <class name="testCases.TC02_Login">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:51:53 EDT" name="openApp" finished-at="2025-08-28T11:51:55 EDT" duration-ms="2061" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:51:55 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:51:59 EDT" duration-ms="3751" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[smitasharma.qa@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[ss*QA123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:51:59 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:01 EDT" duration-ms="1899" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[nona@yahoo.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[QWER!@#$]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:52:01 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:03 EDT" duration-ms="1879" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[sid@cloudberry.services]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Test123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:52:03 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:05 EDT" duration-ms="1958" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[apushedwoman@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[password123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:52:05 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:07 EDT" duration-ms="2111" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[hichamkhachiai@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[hicham123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC02_Login@7642df8f]" started-at="2025-08-28T11:52:07 EDT" name="closeApp" finished-at="2025-08-28T11:52:07 EDT" duration-ms="360" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC02_Login -->
      <class name="testCases.TC05_AddToWishList">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" started-at="2025-08-28T11:51:03 EDT" name="openApp" finished-at="2025-08-28T11:51:07 EDT" duration-ms="4195" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" started-at="2025-08-28T11:51:07 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:20 EDT" duration-ms="13054" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" started-at="2025-08-28T11:51:20 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:30 EDT" duration-ms="10063" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" started-at="2025-08-28T11:51:30 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:40 EDT" duration-ms="10022" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC05_AddToWishList@1e0f9063]" started-at="2025-08-28T11:51:44 EDT" name="closeApp" finished-at="2025-08-28T11:51:45 EDT" duration-ms="713" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC05_AddToWishList -->
      <class name="testCases.TC01_LaunchApplication">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC01_LaunchApplication@5d5f10b2]" started-at="2025-08-28T11:50:33 EDT" name="openApp" finished-at="2025-08-28T11:50:43 EDT" duration-ms="9689" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@5d5f10b2]" started-at="2025-08-28T11:50:43 EDT" name="testLaunchApplication" finished-at="2025-08-28T11:50:44 EDT" duration-ms="976" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLaunchApplication -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC01_LaunchApplication@5d5f10b2]" started-at="2025-08-28T11:50:44 EDT" name="closeApp" finished-at="2025-08-28T11:50:44 EDT" duration-ms="650" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC01_LaunchApplication -->
      <class name="testCases.TC03_AddToCart">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC03_AddToCart@74c79fa2]" started-at="2025-08-28T11:50:44 EDT" name="openApp" finished-at="2025-08-28T11:50:53 EDT" duration-ms="8547" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@74c79fa2]" started-at="2025-08-28T11:50:53 EDT" name="testAddToCart" finished-at="2025-08-28T11:51:01 EDT" duration-ms="7933" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToCart -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC03_AddToCart@74c79fa2]" started-at="2025-08-28T11:51:01 EDT" name="closeApp" finished-at="2025-08-28T11:51:03 EDT" duration-ms="2170" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC03_AddToCart -->
      <class name="testCases.TC06_AddAffiliate">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC06_AddAffiliate@53bd8fca]" started-at="2025-08-28T11:51:45 EDT" name="openApp" finished-at="2025-08-28T11:51:48 EDT" duration-ms="2586" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@53bd8fca]" started-at="2025-08-28T11:51:48 EDT" name="testAddAffiliate" finished-at="2025-08-28T11:51:52 EDT" duration-ms="4702" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddAffiliate -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC06_AddAffiliate@53bd8fca]" started-at="2025-08-28T11:51:52 EDT" name="closeApp" finished-at="2025-08-28T11:51:53 EDT" duration-ms="352" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC06_AddAffiliate -->
      <class name="testCases.TC04_CompletePurchase">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC04_CompletePurchase@3e30646a]" started-at="2025-08-28T11:52:07 EDT" name="openApp" finished-at="2025-08-28T11:52:11 EDT" duration-ms="3794" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[edge]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@3e30646a]" started-at="2025-08-28T11:52:11 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:21 EDT" duration-ms="9723" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC04_CompletePurchase@3e30646a]" started-at="2025-08-28T11:52:21 EDT" name="closeApp" finished-at="2025-08-28T11:52:21 EDT" duration-ms="580" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC04_CompletePurchase -->
    </test> <!-- edgeTest -->
    <test started-at="2025-08-28T11:50:33 EDT" name="firefoxTest" finished-at="2025-08-28T11:52:39 EDT" duration-ms="125895">
      <class name="testCases.TC02_Login">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:06 EDT" name="openApp" finished-at="2025-08-28T11:52:14 EDT" duration-ms="7546" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:15 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:16 EDT" duration-ms="1703" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[smitasharma.qa@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[ss*QA123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:16 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:17 EDT" duration-ms="1185" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[nona@yahoo.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[QWER!@#$]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:17 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:19 EDT" duration-ms="1449" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[sid@cloudberry.services]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Test123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:19 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:21 EDT" duration-ms="1876" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[apushedwoman@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[password123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:21 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:22 EDT" duration-ms="1729" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[hichamkhachiai@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[hicham123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC02_Login@6399551e]" started-at="2025-08-28T11:52:22 EDT" name="closeApp" finished-at="2025-08-28T11:52:24 EDT" duration-ms="1778" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC02_Login -->
      <class name="testCases.TC05_AddToWishList">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" started-at="2025-08-28T11:51:21 EDT" name="openApp" finished-at="2025-08-28T11:51:26 EDT" duration-ms="5767" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" started-at="2025-08-28T11:51:26 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:30 EDT" duration-ms="3513" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" started-at="2025-08-28T11:51:30 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:40 EDT" duration-ms="10113" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" started-at="2025-08-28T11:51:40 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:50 EDT" duration-ms="10128" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC05_AddToWishList@79351f41]" started-at="2025-08-28T11:51:50 EDT" name="closeApp" finished-at="2025-08-28T11:51:52 EDT" duration-ms="1390" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC05_AddToWishList -->
      <class name="testCases.TC01_LaunchApplication">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC01_LaunchApplication@49c6c24f]" started-at="2025-08-28T11:50:33 EDT" name="openApp" finished-at="2025-08-28T11:50:55 EDT" duration-ms="21876" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@49c6c24f]" started-at="2025-08-28T11:50:55 EDT" name="testLaunchApplication" finished-at="2025-08-28T11:50:55 EDT" duration-ms="64" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLaunchApplication -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC01_LaunchApplication@49c6c24f]" started-at="2025-08-28T11:50:55 EDT" name="closeApp" finished-at="2025-08-28T11:50:58 EDT" duration-ms="3187" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC01_LaunchApplication -->
      <class name="testCases.TC03_AddToCart">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC03_AddToCart@6fe1b4fb]" started-at="2025-08-28T11:50:58 EDT" name="openApp" finished-at="2025-08-28T11:51:10 EDT" duration-ms="11860" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@6fe1b4fb]" started-at="2025-08-28T11:51:10 EDT" name="testAddToCart" finished-at="2025-08-28T11:51:18 EDT" duration-ms="7926" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToCart -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC03_AddToCart@6fe1b4fb]" started-at="2025-08-28T11:51:18 EDT" name="closeApp" finished-at="2025-08-28T11:51:21 EDT" duration-ms="2601" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC03_AddToCart -->
      <class name="testCases.TC06_AddAffiliate">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC06_AddAffiliate@1c32386d]" started-at="2025-08-28T11:51:52 EDT" name="openApp" finished-at="2025-08-28T11:52:00 EDT" duration-ms="8641" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@1c32386d]" started-at="2025-08-28T11:52:00 EDT" name="testAddAffiliate" finished-at="2025-08-28T11:52:04 EDT" duration-ms="3977" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddAffiliate -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC06_AddAffiliate@1c32386d]" started-at="2025-08-28T11:52:04 EDT" name="closeApp" finished-at="2025-08-28T11:52:06 EDT" duration-ms="2061" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC06_AddAffiliate -->
      <class name="testCases.TC04_CompletePurchase">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" started-at="2025-08-28T11:52:24 EDT" name="openApp" finished-at="2025-08-28T11:52:29 EDT" duration-ms="4742" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[firefox]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method retried="true" signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" started-at="2025-08-28T11:52:29 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:35 EDT" duration-ms="6241" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method retried="true" signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" started-at="2025-08-28T11:52:35 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:36 EDT" duration-ms="547" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" started-at="2025-08-28T11:52:36 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:36 EDT" duration-ms="531" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC04_CompletePurchase@13d73fa]" started-at="2025-08-28T11:52:36 EDT" name="closeApp" finished-at="2025-08-28T11:52:39 EDT" duration-ms="2363" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC04_CompletePurchase -->
    </test> <!-- firefoxTest -->
    <test started-at="2025-08-28T11:50:33 EDT" name="chromeTest" finished-at="2025-08-28T11:52:58 EDT" duration-ms="145051">
      <class name="testCases.TC02_Login">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:51:49 EDT" name="openApp" finished-at="2025-08-28T11:51:51 EDT" duration-ms="2556" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:51:54 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:51:58 EDT" duration-ms="4277" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[smitasharma.qa@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[ss*QA123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:51:58 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:00 EDT" duration-ms="2185" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[nona@yahoo.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[QWER!@#$]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:52:00 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:02 EDT" duration-ms="2174" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[sid@cloudberry.services]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Test123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:52:02 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:04 EDT" duration-ms="1851" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[apushedwoman@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[password123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method signature="testLogin(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:52:04 EDT" name="testLogin" data-provider="LoginData" finished-at="2025-08-28T11:52:06 EDT" duration-ms="2279" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[hichamkhachiai@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[hicham123]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogin -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC02_Login@2f40e5db]" started-at="2025-08-28T11:52:06 EDT" name="closeApp" finished-at="2025-08-28T11:52:07 EDT" duration-ms="1073" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC02_Login -->
      <class name="testCases.TC05_AddToWishList">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" started-at="2025-08-28T11:50:59 EDT" name="openApp" finished-at="2025-08-28T11:51:03 EDT" duration-ms="3673" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" started-at="2025-08-28T11:51:03 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:19 EDT" duration-ms="15639" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method retried="true" signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" started-at="2025-08-28T11:51:19 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:29 EDT" duration-ms="10189" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method signature="testAddToWishList()[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" started-at="2025-08-28T11:51:29 EDT" name="testAddToWishList" finished-at="2025-08-28T11:51:39 EDT" duration-ms="10356" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC05_AddToWishList.testAddToWishList(TC05_AddToWishList.java:68)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToWishList -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC05_AddToWishList@e8df99a]" started-at="2025-08-28T11:51:39 EDT" name="closeApp" finished-at="2025-08-28T11:51:41 EDT" duration-ms="1444" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC05_AddToWishList -->
      <class name="testCases.TC01_LaunchApplication">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC01_LaunchApplication@22875539]" started-at="2025-08-28T11:50:33 EDT" name="openApp" finished-at="2025-08-28T11:50:43 EDT" duration-ms="9713" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testLaunchApplication()[pri:0, instance:testCases.TC01_LaunchApplication@22875539]" started-at="2025-08-28T11:50:43 EDT" name="testLaunchApplication" finished-at="2025-08-28T11:50:43 EDT" duration-ms="167" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLaunchApplication -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC01_LaunchApplication@22875539]" started-at="2025-08-28T11:50:43 EDT" name="closeApp" finished-at="2025-08-28T11:50:43 EDT" duration-ms="501" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC01_LaunchApplication -->
      <class name="testCases.TC03_AddToCart">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC03_AddToCart@e54303]" started-at="2025-08-28T11:50:43 EDT" name="openApp" finished-at="2025-08-28T11:50:47 EDT" duration-ms="3959" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddToCart()[pri:0, instance:testCases.TC03_AddToCart@e54303]" started-at="2025-08-28T11:50:47 EDT" name="testAddToCart" finished-at="2025-08-28T11:50:56 EDT" duration-ms="8906" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddToCart -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC03_AddToCart@e54303]" started-at="2025-08-28T11:50:56 EDT" name="closeApp" finished-at="2025-08-28T11:50:59 EDT" duration-ms="3030" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC03_AddToCart -->
      <class name="testCases.TC06_AddAffiliate">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC06_AddAffiliate@2dc995f4]" started-at="2025-08-28T11:51:41 EDT" name="openApp" finished-at="2025-08-28T11:51:44 EDT" duration-ms="2920" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method signature="testAddAffiliate()[pri:0, instance:testCases.TC06_AddAffiliate@2dc995f4]" started-at="2025-08-28T11:51:44 EDT" name="testAddAffiliate" finished-at="2025-08-28T11:51:49 EDT" duration-ms="4760" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddAffiliate -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC06_AddAffiliate@2dc995f4]" started-at="2025-08-28T11:51:49 EDT" name="closeApp" finished-at="2025-08-28T11:51:49 EDT" duration-ms="187" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC06_AddAffiliate -->
      <class name="testCases.TC04_CompletePurchase">
        <test-method is-config="true" signature="openApp(java.lang.String,java.lang.String)[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" started-at="2025-08-28T11:52:07 EDT" name="openApp" finished-at="2025-08-28T11:52:11 EDT" duration-ms="3944" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[windows]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openApp -->
        <test-method retried="true" signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" started-at="2025-08-28T11:52:11 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:31 EDT" duration-ms="19827" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method retried="true" signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" started-at="2025-08-28T11:52:31 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:46 EDT" duration-ms="14314" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method signature="testCompletePurchase()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" started-at="2025-08-28T11:52:46 EDT" name="testCompletePurchase" finished-at="2025-08-28T11:52:58 EDT" duration-ms="12059" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Test failed due to unexpected exception.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Test failed due to unexpected exception.
at org.testng.Assert.fail(Assert.java:111)
at testCases.TC04_CompletePurchase.testCompletePurchase(TC04_CompletePurchase.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:463)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1575)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCompletePurchase -->
        <test-method is-config="true" signature="closeApp()[pri:0, instance:testCases.TC04_CompletePurchase@517566b]" started-at="2025-08-28T11:52:58 EDT" name="closeApp" finished-at="2025-08-28T11:52:58 EDT" duration-ms="170" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeApp -->
      </class> <!-- testCases.TC04_CompletePurchase -->
    </test> <!-- chromeTest -->
  </suite> <!-- Suite -->
</testng-results>
